elements/playerbufftimers.lua

```diff
@@ -0,0 +1,290 @@
+	* index - index at which the timer was created
+	--]]
+	if(element.PostCreateTimer) then
+		element:PostCreateTimer(timer, index)
+	end
+
+	return timer
+end
+
+-- local function UnitPowerBarTimerInfo(unit, index)
+-- 	if(index > 5) then return end
+-- 	local duration = math.random(20, 60)
+-- 	return duration, GetTime() + duration, 84, 101871
+-- end
+
+local function Update(self, event)
+	local element = self.PlayerBuffTimers
+
+	--[[ Callback: PlayerBuffTimers:PreUpdate()
+	Called before the element has been updated.
+
+	* self  - the PlayerBuffTimers widget
+	--]]
+	if(element.PreUpdate) then element:PreUpdate() end
+
+	local index = 1
+	local duration, expiration, barID, auraID = UnitPowerBarTimerInfo('player', index)
+	while(barID) do
+		local timer = element[index]
+		if(not timer) then
+			--[[ Override: PlayerBuffTimers:CreateTimer(index)
+			Used to create and return a new timer widget.
+
+			* self  - the PlayerBuffTimers widget
+			* index - index at which the timer should be created
+			--]]
+			timer = (element.CreateTimer or CreateTimer)(element, index)
+			element[#element + 1] = timer
+		end
+
+		timer.powerName, timer.powerTooltip = select(11, GetAlternatePowerInfoByID(barID))
+
+		--[[ Override: timer:UpdateTimer(duration, expiration, barID, auraID)
+		Used to update the timer attributes.
+
+		* self       - widget holding the timer to be updated
+		* duration   - total timer duration
+		* expiration - time at which the timer expires. Can be compared to `GetTime()`
+		* barID      - alternate power id of the timer
+		* auraID     - spellid of the timer aura
+		--]]
+		timer:UpdateTimer(duration, expiration, barID, auraID)
+		timer.show = true
+
+		index = index + 1
+		duration, expiration, barID, auraID = UnitPowerBarTimerInfo('player', index)
+	end
+
+	for i = 1, #element do
+		local timer = element[i]
+		if(timer.show) then
+			timer.show = nil
+			timer:Show()
+		else
+			timer:Hide()
+		end
+	end
+
+	--[[ Override: PlayerBuffTimers:SetPosition()
+	Used to (re-)anchor the timers.
+	Called every time a new timer is shown or an old one expires.
+
+	* self - the PlayerBuffTimers widget
+	--]]
+	(element.SetPosition or SetPosition)(element)
+
+	--[[ Callback: PlayerBuffTimers:PostUpdate()
+	Called after all timers have been updated.
+
+	* self - the PlayerBuffTimers widget
+	--]]
+	if(element.PostUpdate) then
+		element:PostUpdate()
+	end
+end
+
+local function Path(self, ...)
+	--[[ Override: PlayerBuffTimers:Override(...)
+	Used to completely override the internal update function.
+
+	* self - the PlayerBuffTimers widget
+	* ...  - the event and the arguments that accompany it
+	--]]
+	return (self.PlayerBuffTimers.Override or Update)(self, ...)
+end
+
+local function ForceUpdate(element)
+	return Path(element.__owner, 'ForceUpdate', 'player')
+end
+
+local function Enable(self, unit)
+	if(unit ~= 'player') then return end
+	local element = self.PlayerBuffTimers
+	if(not element) then return end
```

Comment:
This is just me yes - avoid table access if you don't need it and save indents if you can. Can change for consistency, not a problem.